nix/services/devshell.nix

```diff
@@ -0,0 +1,53 @@
+{ config, pkgs, lib, ... }:
+let
+  inherit (lib) types;
+in
+{
+  options.services.outputs = {
+    enabledServices = lib.mkOption {
+      type = types.listOf (types.listOf types.str);
+      readOnly = true;
+      description = ''
+        List of names of enabled services.
+
+        Each item will be of the form [ "<service>" "<instance-name>" ]
```

Comment:
Why not `{ service = "<service">; instance = "<instance-name>"; }`? Or even `{ service1 = [ instance 1 ... ]; ... }`.

Types should not be designed based on the logic of how they get used (`attrByPath`), but rather by clarity of purpose. Also the lists can represent illegal shapes (`["foo" "bar" "qux"]`)  than a structured attrset here.

Reply:
How about https://github.com/juspay/services-flake/pull/355/commits/b92546c0c823180c03e3a331594e5cdf9d05d8fe?